FIAT/functional.py

```diff
@@ -336,22 +339,214 @@ def __init__(self, ref_el, facet_no, Q, f_at_qpts):
                             {}, dpt_dict, "IntegralMomentOfNormalDerivative")
 
 
+class IntegralLegendreDirectionalMoment(Functional):
+    """Momement of v.s against a Legendre polynomial over an edge"""
+    def __init__(self, cell, s, entity, mom_deg, comp_deg, nm=""):
+        sd = cell.get_spatial_dimension()
+        assert sd == 2
+        shp = (sd,)
+        quadpoints = comp_deg + 1
+        Q = GaussLegendreQuadratureLineRule(interval(), quadpoints)
+        legendre = numpy.polynomial.legendre.legval(2*Q.get_points()-1, [0]*mom_deg + [1])
+        f_at_qpts = numpy.array([s*legendre[i] for i in range(quadpoints)])
+        fmap = cell.get_entity_transform(sd-1, entity)
+        mappedqpts = [fmap(pt) for pt in Q.get_points()]
+        mappedQ = QuadratureRule(cell, mappedqpts, Q.get_weights())
+        qwts = mappedQ.wts
+        qpts = mappedQ.pts
+
+        pt_dict = OrderedDict()
+
+        for k in range(len(qpts)):
+            pt_cur = tuple(qpts[k])
+            pt_dict[pt_cur] = [(qwts[k] * f_at_qpts[k, i], (i,))
+                               for i in range(2)]
+
+        Functional.__init__(self, cell, shp, pt_dict, {}, nm)
```

Comment:
This is very old style, `super().__init__(...)` is preferable.

Reply:
Fixed (throughout `functional.py`).